Source/WebCore/bindings/js/JSDOMConvertResult.h

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright (C) 2024 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#include <functional>
+#include <type_traits>
+#include <utility>
+#include <wtf/Expected.h>
+
+namespace WebCore {
+
+template<typename T> struct Converter;
+
+// Result a conversion from JSValue -> Implementation.
+template<typename IDL> class ConversionResult;
+
+// Token used to indicate that a conversion from JSValue -> Implementation has failed.
+struct ConversionResultException { };
+
+namespace Detail {
+
+template<typename T>
+struct ConversionResultStorage {
+    using ReturnType = T;
+    using Type = T;
+
+    ConversionResultStorage(ConversionResultException token) : value(makeUnexpected(token)) { }
+    ConversionResultStorage(const Type& value) : value(value) { }
+    ConversionResultStorage(Type&& value) : value(WTFMove(value)) { }
+
+    template<typename U>
+    ConversionResultStorage(ConversionResultStorage<U>&& other)
+        : value([&]() -> Expected<Type, ConversionResultException> {
+            if (other.hasException())
+                return makeUnexpected(ConversionResultException());
+            else
```

Comment:
Nit: It seems like we don't need to have else here.

Reply:
Fixed.